### PR TITLE
Tls and initial disconnected support

### PR DESCRIPTION
**Pull request by aleskandro**
The SystemConfigSyncer is a thread-safe singleton object responsible for keeping (1) the registries.conf, (2) the policy.json, and (3) the per-host certificates authority to store in the certs.d docker config folder.
When instantiated as a singleton, it also registers handlers for:
1. The openshift-image-registry/image-registry-certificates configmap, storing the per-registry custom certificates authorities
2. The image.config.openshift.io/cluster object, storing the allowed, blocked and insecure registries

[TODO] The ICSP and IDMS controllers will also use the methods defined in this package to marshall the image registries mirror configuration into the registries.conf files

This PR also fixes the SystemContext for non manifest-list inspection

### CODE REVIEW

File: pkg/system_config/syncer.go [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+		policyConfContent:     defaultPolicyConf(),
+		registryCertTuples:    []registryCertTuple{},
+		ch:                    make(chan bool),
+	}
+	go ic.syncer()
+	ctx := context.Background()
+	err := core.NewSingleObjectEventHandler[*v1.ConfigMap, *v1.ConfigMapList](ctx,
+		"image-registry-certificates", "openshift-image-registry",
+		time.Hour, func(et watch.EventType, cm *v1.ConfigMap) {
+			if et == watch.Deleted || et == watch.Bookmark {
```
Comment (Prashanth684) [inline pkg/system_config/syncer.go:165]: can these ever be deleted/absent? should we account for that

Comment (aleskandro) [reply pkg/system_config/syncer.go:165]: Yep, thanks! Bad copy and paste from the other handler for the global pull secret. I'll add the methods already available in the syncer.

File: pkg/system_config/types.go [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+		Type: "reject",
+	}
+}
+
+type policyEntry struct {
+	Type string `json:"type"`
+}
+
+func writeTomlFile(path string, data interface{}) error {
+	createBaseDir(path)
```
Comment (Prashanth684) [inline pkg/system_config/types.go:169]: any possibility to have these files as mem mapped files too?

Comment (aleskandro) [reply pkg/system_config/types.go:169]: I initially thought about that too. 
However, the (asynchronous events based) sources are several in this case,  and we still need to take care of concurrency. Also, while the ones we generate for each pod have the potential to become a lot and to strongly impact i/o, the ones deriving from these handlers should be much more stable in the cluster lifetime. Moreover, the custom CAs require a precise FS structure.

Therefore, I decided not to follow that path in this case.. Wdyt?

File: pkg/system_config/types.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+	// Setting the blocked, allowed and insecure fields to nil will cause them to be omitted from the output
+	Blocked  *bool `toml:"blocked"`
+	Allowed  *bool `toml:"allowed"`
+	Insecure *bool `toml:"insecure"`
+}
+
+// defaultRegistriesConf returns a default registriesConf object
+func defaultRegistriesConf() registriesConf {
+	return registriesConf{
+		UnqualifiedSearchRegistries: []string{"registry.access.redhat.com", "docker.io"},
```
Comment (Prashanth684) [inline pkg/system_config/types.go:94]: are these defaults present in the node itself for all scenarios?

File: pkg/system_config/syncer.go (hunk 1 above)
Comment (aleskandro) [reply pkg/system_config/syncer.go:165]: mmm relooking at this, I think the events should not be handled by us as they will trigger a reconcile in other operators that will re-create the objects, hence triggering this handler again for the correct event to handle

File: pkg/system_config/types.go (hunk 3 above)
Comment (aleskandro) [reply pkg/system_config/types.go:94]: yep.

**Comment by aleskandro** [comment]
merging for now. The ICSP and IDMS controllers will be implemented in a future PR.

**Pull request merged**